lib/sanbase/accounts/user/user_email.ex

```
@@ -154,6 +154,16 @@ defmodule Sanbase.Accounts.User.Email do
       |> Map.merge(%{token: user.email_token, email: user.email})
       |> Map.merge(Map.take(args, [:subscribe_to_weekly_newsletter]))
 
+    query_map =
+      if valid_redirect_url?(args[:success_redirect_url]),
+        do: Map.put(query_map, :success_redirect_url, args[:success_redirect_url]),
+        else: query_map
+
+    query_map =
+      if valid_redirect_url?(args[:fail_redirect_url]),
+        do: Map.put(query_map, :fail_redirect_url, args[:fail_redirect_url]),
+        else: query_map
+
```

Hm, what if you do in a `with` statement?
```elixir
with {:ok, query_map} <- maybe_add_url(query_map, args, :success_redirect_url)
         {:ok, query_map} <- maybe_add_url(query_map, args, :fail_redirect_url) dO
         ...
```
so we don't hide errors for invalid URLs?

---

you mean to return the error to the frontend if of one of redirect urls is malformed ?